### PR TITLE
Use an entrypoint script for the Dockerfile

### LINKED ISSUE #72: Implement way to provide list of plugin repositories
Currently, there is no way to let the Dockerfile know what plugin repositories to install, which are then available to the OTEAPI service.

If one should not have to build new images over and over, I suggest to implement this in the `CMD/ENTRYPOINT` command in the Dockerfile. Note, this/these commands can already be overwritten through a docker-compose file, meaning there is, in theory, no need to change the Dockerfile or Docker images currently released, but rather update a docker-file appropriately.

### PR DESCRIPTION
**Pull request by CasperWA**
Closes #72 

This enables a more sophisticated way of installing plugin packages.

To accommodate the `development` target, when installing a custom/editable/local version of `oteapi-core`, and list of package constraints is made on-the-fly.

---

To do:
- [x] Document method of installing plugin packages.
- [ ] _(To be discussed and agreed upon in the comments of this PR)_ Determine proper way of creating the list of package constraints (the `constraints.txt` file).

### CODE REVIEW

**Comment by CasperWA** [comment]
Concerning the `constraints.txt` file, I have updated the construction method to the following: Copy `requirements.txt` and change only the version for `oteapi-core` according to the currently installed version (this will only ever change if a custom `oteapi-core` version is installed during the `development` target that has a different version number than what is specified in `requirements.txt`).

File: Dockerfile [hunk 1]
```diff
@@ -36,11 +36,13 @@ RUN pip install --trusted-host pypi.org --trusted-host files.pythonhosted.org -r
 
 # Run app with reload option
 EXPOSE 8080
-CMD if [ "${PATH_TO_OTEAPI_CORE}" != "/dev/null" ] && [ -n "${PATH_TO_OTEAPI_CORE}" ]; then pip install -U --force-reinstall -e /oteapi_core; fi && hypercorn asgi:app --bind 0.0.0.0:8080 --reload --debug --log-level debug
+CMD if [ "${PATH_TO_OTEAPI_CORE}" != "/dev/null" ] && [ -n "${PATH_TO_OTEAPI_CORE}" ]; then \
```
Comment (jesper-friis) [inline Dockerfile:39]: Checking `PATH_TO_OTEAPI_CORE` against `/dev/null` seems quite arbitrary? Is there any reason to expect a user to set `PATH_TO_OTEAPI_CORE` to `/dev/null`?

File: README.md [hunk 2]
```diff
@@ -92,10 +94,10 @@ docker run \
     --env OTEAPI_REDIS_TYPE=redis \
     --env OTEAPI_REDIS_HOST=redis \
     --env OTEAPI_REDIS_PORT=6379 \
-    ontotrans/oteapi:latest
+    emmc-asbl/oteapi:latest
 ```
 
-Open the following URL in a browser [http://localhost:80/redoc](http://localhost:80/redoc).
+Open the following URL in a browser [http://localhost/redoc](http://localhost:80/redoc).
```
Comment (jesper-friis) [inline README.md:100]: According to Dockerfile and entrypoint.sh runs production also on port 8080
```suggestion
Open the following URL in a browser [http://localhost:8080/redoc](http://localhost:8080/redoc).
```

File: README.md [hunk 3]
```diff
@@ -127,24 +130,25 @@ docker-compose -f docker-compose_dev.yml up -d  # Run the OTE Services (detached
 Note that default values will be used if certain environment variables are not present.
 To inspect which environment variables can be specified, please inspect the [Docker Compose file](docker-compose_dev.yml).
 
-This Docker Compose file will use your local files for the application, meaning updates in your local files (under `app/`) should be reflected in the running application after hypercorn reloads.
+This Docker Compose file will use your local files for the application, meaning updates in your local files (under `app/`) should be reflected in the running application upon storing the changes to disk, after hypercorn reloads.
 You can go one step further and use your local files also for the `oteapi-core` repository by specifying the `PATH_TO_OTEAPI_CORE` environment variable:
```
Comment (jesper-friis) [inline README.md:135]: It would probably be even more useful if one could specify path to local versions of plugins one want to include.

**Review by jesper-friis — COMMENTED** [review]
Feel free to merge when the port number in the readme file is corrected.

File: README.md (hunk 2 above)
Comment (CasperWA) [reply README.md:100]: Are you sure? I think for this particular example it should map to port `80`?

File: README.md (hunk 3 above)
Comment (CasperWA) [reply README.md:135]: Hmmm... yes that would also be very beneficial!
Let's make a separate issue for it perhaps, as this is a developer thing, while this PR implements something general that should also be used in production.

**Comment by CasperWA** [comment]
> Feel free to merge when the port number in the readme file is corrected.

Are you sure it needs to be? I think for that particular example we're running in production, mapping to port `80`.

**Comment by jesper-friis** [comment]
> >Feel free to merge when the port number in the readme file is corrected.
> 
> Are you sure it needs to be? I think for that particular example we're running in production, mapping to port 80.

But isn't the production also using port 8080? It seems so in the entry_point.sh file

**Comment by CasperWA** [comment]
> > > Feel free to merge when the port number in the readme file is corrected.
> > 
> > 
> > Are you sure it needs to be? I think for that particular example we're running in production, mapping to port 80.
> 
> But isn't the production also using port 8080? It seems so in the entry_point.sh file

Sure. Within the docker container. But that internal port is mapped to `80` when using the `--publish 80:8080` option for the `docker run` command - or setting the environment variable `PORT` to `80` when using docker-compose.

File: README.md (hunk 2 above)
Comment (CasperWA) [reply README.md:100]: For production the README maps the `8080` port within the docker container to port `80` on the host.

File: README.md (hunk 3 above)
Comment (CasperWA) [reply README.md:135]: Created issue #76

**Review by jesper-friis — APPROVED** [review]

**Comment by jesper-friis** [comment]
> > > > Feel free to merge when the port number in the readme file is corrected.
> > > 
> > > 
> > > Are you sure it needs to be? I think for that particular example we're running in production, mapping to port 80.
> > 
> > 
> > But isn't the production also using port 8080? It seems so in the entry_point.sh file
> 
> Sure. Within the docker container. But that internal port is mapped to `80` on the local host when using the `--publish 80:8080` option for the `docker run` command - or setting the environment variable `PORT` to `80` when using docker-compose.

I see, then I approve

**Pull request merged**